.github/workflows/release.yaml

```diff
@@ -0,0 +1,69 @@
+# When pushing to main, we run checks and then release.
+
+name: Check and Release
+on:
+  push:
+    branches:
+    - main
+permissions:
+  contents: read
+env:
+  GENERATED_PACKAGE_DIR: dist
+  ARTIFACT_NAME: artifact-ubuntu-latest-3.10
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: write
+    secrets: inherit
+
+  provenance:
+    needs: build
+    permissions:
+      actions: read
+      id-token: write
+      contents: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@d995948e8d53cc639c0d3ef69db31dbc243519c4
+    with:
+      base64-subjects: ${{ needs.build.outputs.artifacts-sha256 }}
+
+  release:
+    if: "!startsWith(github.event.commits[0].message, 'bump:')"
+    needs: [build, provenance]
```

Comment:
Isn’t it enough to use
```yaml
needs: provenance
```
because the `provenance` job needs the `build` job to succeed?

Reply:
> Isn’t it enough to use
> 
> ```yaml
> needs: provenance
> ```
> 
> because the `provenance` job needs the `build` job to succeed?

No because we need the access the output from both jobs and if we don't add both to `needs`, it will fail.

Reply:
Huh 😳 I thought that failure cascades…

Reply:
> Huh flushed I thought that failure cascades…

The failure cascades, but the outputs are piped only when jobs are specified in `needs` attribute.

Reply:
Ah!